app/src/main/java/com/codesoom/assignment/handlers/TaskHttpHandler.java

```diff
@@ -0,0 +1,163 @@
+        outputStream.flush();
+        outputStream.close();
+    }
+
+    private ResponseDto handleRequest(String httpMethod, String requestURI, String requestBody) throws IOException {
+        if (PATH.equals(requestURI) && HttpMethod.GET.name().equals(httpMethod)) {
+            return getTasks();
+        }
+        if (PATH.equals(requestURI) && HttpMethod.POST.name().equals(httpMethod)) {
+            return createTask(toTask(requestBody));
+        }
+        if (isMatch(requestURI) && HttpMethod.GET.name().equals(httpMethod)) {
+            return getTask(getPathVariable(requestURI));
+        }
+        if (isMatch(requestURI) && HttpMethod.PATCH.name().equals(httpMethod)) {
+            return updateTask(getPathVariable(requestURI), toTask(requestBody));
+        }
+        if (isMatch(requestURI) && HttpMethod.DELETE.name().equals(httpMethod)) {
+            return deleteTask(getPathVariable(requestURI));
+        }
+
+        return handleError(HTTP_STATUS_NOT_FOUND, "Not Found");
+    }
+
+
+    private ResponseDto getTasks() throws IOException {
+        return new ResponseDto(HTTP_STATUS_OK, tasksToJSON(taskService.getTasks()));
+    }
+
+    private ResponseDto createTask(TaskDto requestDto) throws IOException {
+        final Task newTask = taskService.addTask(requestDto);
+        return new ResponseDto(HTTP_STATUS_CREATE, taskToJSON(newTask));
+    }
+
+    private ResponseDto getTask(Long id) throws IOException {
+        final Task findTask = taskService.findTaskById(id);
+        return new ResponseDto(HTTP_STATUS_OK, taskToJSON(findTask));
+    }
+
+    private ResponseDto updateTask(Long id, TaskDto requestDto) throws IOException {
+        final Task updatedTask = taskService.updateTaskById(id, requestDto);
+        return new ResponseDto(HTTP_STATUS_OK, taskToJSON(updatedTask));
+    }
+
+    private ResponseDto deleteTask(Long id) {
+        taskService.deleteTaskById(id);
+        return new ResponseDto(HTTP_STATUS_OK, "");
+    }
+
+    private ResponseDto handleError(int httpStatus, String message) {
+        return new ResponseDto(httpStatus, message);
+    }
+
+    /**
+     * 모든 Task 목록을 JSON 형식으로 반환한다.
+     * */
+    private String tasksToJSON(List<Task> tasks) throws IOException {
+        OutputStream outputStream = new ByteArrayOutputStream();
+        objectMapper.writeValue(outputStream, tasks);
+
+        return outputStream.toString();
+    }
+
+    /**
+     * 하나의 Task 객체를 JSON 형식으로 반환한다.
+     * */
+    private String taskToJSON(Task task) throws IOException {
+        OutputStream outputStream = new ByteArrayOutputStream();
+        objectMapper.writeValue(outputStream, task);
+
+        return outputStream.toString();
+    }
+
+    /**
+     * 요청받은 할 일을 Task 객체로 변환하여 반환한다.
+     * */
+    private TaskDto toTask(String content) {
+        System.out.println("content = " + content);
+        Pattern pattern = Pattern.compile("\\{[\\n\\t\\s]*\"title\"[\\n\\t\\s]*:[\\n\\t\\s]*\"(.*?)\"[\\n\\t\\s]*}");
```

Comment:
`\s` 클래스에 `\n`과 `\t`가 포함되어 있어서 이렇게 생략해주셔도 됩니다.
```suggestion
        Pattern pattern = Pattern.compile("\\{\\s*\"title\"\\s*:\\s*\"(.*?)\"[\\s]*}");
```

Reply:
정규식으로 한 번 파싱해봤는데, 많은 공부가 필요할 것 같네요😂
`(.*?)`사용에 대해 좀 더 고민해보겠습니다.